### PR TITLE
KIND: Fix output path for api conversation generation + Fix webhook ca bundle injection for flow-collector CRD

### PR DESCRIPTION
**Pull request by astoycos**
This PR fixes two bugs:  

1. 
There are still some webhook issues when running on KIND... specifically I spin things up with 

1. `IMG="quay.io/bpfd/network-observability-operator:test" make image-build image-push`
2. `make local-deploy`
3. `IMG="quay.io/bpfd/network-observability-operator:test" make deploy-kind` 
4. `make deploy-sample-cr` 

I run into 
```
==> Deploy sample CR
kubectl apply -f ./config/samples/flows_v1beta1_flowcollector.yaml || true
Error from server: error when creating "./config/samples/flows_v1beta1_flowcollector.yaml": conversion webhook for flows.netobserv.io/v1beta1, Kind=FlowCollector failed: Post "https://netobserv-webhook-service.netobserv.svc:443/convert?timeout=30s": x509: certificate signed by unknown authority
```

operator Logs: 

```
2023/03/02 06:33:07 http: TLS handshake error from 10.244.0.1:3302: remote error: tls: bad certificate
```

This PR fixes the aforementioned bug by ensuring we inject the webhook ca-bundle into the flow-collector CRD definition. 

2. 
The makefile logic to detect weather or not the cloned code is in gopath is flawed....specifically 

```
# Set --output-base for conversion-gen if we are not within GOPATH
ifneq ($(abspath $(ROOT_DIR)),$(shell go env GOPATH)/src/network-observability-operator)
	CONVERSION_GEN_OUTPUT_BASE := --output-base=$(ROOT_DIR)
else
	export GOPATH := $(shell go env GOPATH)
endif
```
Assumes that you specifically cloned the code to `$GOPATH/src/network-observability-operator` which probably isn’t the case for everyone

for example on my setup the repo lives at `/home/astoycos/go/src/github.com/netobserv/network-observability-operator` so although it’s in my GOPATH we still set --output-base here when we don’t need to.  Ultimately I was seeing `zz_generated.conversion.go` get put in the wrong spot because of this

The fix just makes sure that the $GOPATH is in wherever the code lives (i.e $ROOT_DIR)

### CODE REVIEW

**Comment by msherif1234** [comment]
@astoycos pls make sure u update to main latest all webhook issues should have been addressed as well as kind issues

File: Makefile [hunk 1]
```diff
@@ -217,7 +217,8 @@ generate-go-conversions: $(CONVERSION_GEN) ## Run all generate-go-conversions
 		$(CONVERSION_GEN) \
 		--input-dirs=./api/v1alpha1 \
 		--build-tag=ignore_autogenerated_core \
-		--output-file-base=zz_generated.conversion $(CONVERSION_GEN_OUTPUT_BASE)\
```
Comment (msherif1234) [inline Makefile:220]: nit why split the line seems unneeded change ?

**Comment by astoycos** [comment]
/assign @msherif1234 

This should be good to go now!

File: config/kubernetes/manager_webhook_patch.yaml [hunk 2]
```diff
@@ -21,3 +21,11 @@ spec:
         secret:
           defaultMode: 420
           secretName: webhook-server-cert
+---
+# The following patch adds a directive for certmanager to inject CA into the CRD
```
Comment (msherif1234) [inline config/kubernetes/manager_webhook_patch.yaml:25]: isn't better/cleaner to have this in its own patch file and let `kustomization` patch it ?

**Review by msherif1234 — COMMENTED** [review]
overall looks good thanks for fixing KinD env, just left a couple of nits

**Comment by msherif1234** [comment]
also there is no longer version updates so pls adjust the title and PR description just indicating for kind we need to have cert-mage CA or something similar

File: Makefile (hunk 1 above)
Comment (astoycos) [reply Makefile:220]: Just code cleanliness tbh :)

File: config/kubernetes/manager_webhook_patch.yaml (hunk 2 above)
Comment (astoycos) [reply config/kubernetes/manager_webhook_patch.yaml:25]: Sure, it's just a preference thing I can do that if you'd like, I wish I could just point to `../crd/patches/cainjection_in_flowcollectors.yaml` but it won't let me point to that file from the `kubernetes/` directory

Comment (msherif1234) [reply config/kubernetes/manager_webhook_patch.yaml:25]: just wanted to keep the original sdk yamls as close as the original auto generated ones so any future interactions with sdk-operator will be smooth

Comment (msherif1234) [reply config/kubernetes/manager_webhook_patch.yaml:25]: I can do that later if needed

Comment (astoycos) [reply config/kubernetes/manager_webhook_patch.yaml:25]: All good I can do it now

Comment (astoycos) [reply config/kubernetes/manager_webhook_patch.yaml]: Done

Comment (msherif1234) [reply config/kubernetes/manager_webhook_patch.yaml]: Thanks!
/lgtm

**Comment by jotak** [comment]
tested and lgtm, thanks @astoycos !
/approve

**Pull request merged**